app/assets/javascripts/components/donut_chart.js.jsx

```diff
@@ -0,0 +1,87 @@
+var DonutChart = React.createClass({
+  displayName: 'DonutChart',
+  /*
+     Creates a donut chart using c3 -
+     see https://www.patternfly.org/patterns/donut-chart/
+
+     [+id+] - ID for the element where the chart is rendered
+     [+className] - CSS class(es) to be applied to the div where the chart is rendered
+     [+columns+] - Array of tuples containing label, value. Values must sum 100%
+     e.g. ['success', 40], ['failed', 60]
+     [+groups+] - Set groups for the data for stacking.
+     e.g. [['dataa', 'datab'],['datac']]
+     [+colors+] - Array of HEX color codes for each group.
+     Defaults to Patternfly palette.
+  */
+
+  propTypes: {
+    className: React.PropTypes.string,
+    colors: React.PropTypes.arrayOf(React.PropTypes.string),
+    columns: React.PropTypes.arrayOf(React.PropTypes.array).isRequired,
+    groups: React.PropTypes.arrayOf(React.PropTypes.array),
+    id: React.PropTypes.string.isRequired
+  },
+
+  getDefaultProps: function() {
+    return {
+      className: 'col-md-3',
+      colors: ['#3F9C35', '#C00', '#D1D1D1', '#EC7A08']
+    };
+  },
+
+  componentDidMount: function() {
+    this.generateDonutChart();
```

Comment:
After the `divs` in `render` have been rendered, we call the function to put the chart in place. 

It's a common pattern in React to do this in 2 steps when the component uses d3, jquery, etc.. as these libraries act with the real DOM, while React only deals with a virtual DOM.